app/src/main/java/ru/spbhse/pocketmagic/DatabaseHelper.java

```diff
@@ -0,0 +1,100 @@
+package ru.spbhse.pocketmagic;
+
+import android.content.Context;
+import android.database.SQLException;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class DatabaseHelper extends SQLiteOpenHelper {
+    private static String DB_NAME = "spells.db";
+    private static String DB_PATH = "";
+    private static final int DB_VERSION = 15;
+
+    private SQLiteDatabase mDataBase;
+    private final Context mContext;
+    private boolean mNeedUpdate = false;
+
+    public DatabaseHelper(Context context) {
+        super(context, DB_NAME, null, DB_VERSION);
+        if (android.os.Build.VERSION.SDK_INT >= 17)
+            DB_PATH = context.getApplicationInfo().dataDir + "/databases/";
+        else
+            DB_PATH = "/data/data/" + context.getPackageName() + "/databases/";
+        this.mContext = context;
+
+        copyDataBase();
+
+        this.getReadableDatabase();
+    }
+
+    public void updateDataBase() throws IOException {
+        if (mNeedUpdate) {
+            File dbFile = new File(DB_PATH + DB_NAME);
+            if (dbFile.exists())
+                dbFile.delete();
+
+            copyDataBase();
+
+            mNeedUpdate = false;
+        }
+    }
+
+    private boolean checkDataBase() {
+        File dbFile = new File(DB_PATH + DB_NAME);
+        return dbFile.exists();
+    }
+
+    private void copyDataBase() {
+        if (!checkDataBase()) {
+            this.getReadableDatabase();
+            this.close();
+            try {
+                copyDBFile();
+            } catch (IOException mIOException) {
+                throw new Error("ErrorCopyingDataBase");
+            }
+        }
+    }
+
+    private void copyDBFile() throws IOException {
+        InputStream mInput = mContext.getAssets().open(DB_NAME);
+        //InputStream mInput = mContext.getResources().openRawResource(R.raw.info);
+        OutputStream mOutput = new FileOutputStream(DB_PATH + DB_NAME);
+        byte[] mBuffer = new byte[1024];
+        int mLength;
+        while ((mLength = mInput.read(mBuffer)) > 0)
+            mOutput.write(mBuffer, 0, mLength);
+        mOutput.flush();
```

Comment:
close() should already do flush(). This is redundant.

Reply:
Fixed